### PR TITLE
feat(VCalendar): add new component

### PR DESCRIPTION
**Pull request by johnleider**
#### Replacement for #4846
The beginnings of a Calendar component for Vuetify.
You can tinker with the component by copying [this code](https://gist.github.com/ClickerMonkey/b9ffa574b54c8f9b94207df4d06f7804) into Playground.vue

## Motivation and Context
This introduces the v-calendar and v-calendar-daily component. Once the other v-calendar components are done v-calendar will be updated.

## How Has This Been Tested?
Visually on various browsers with the above Playground code.

## Markup:

https://gist.github.com/ClickerMonkey/b9ffa574b54c8f9b94207df4d06f7804

updated gist (clearing lint errors)
https://gist.github.com/MajesticPotatoe/45b980733a21e0491ec928c0026eb4bb

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes. (@sh7dm)

### CODE REVIEW

File: packages/api-generator/src/map.js [hunk 1]
```diff
@@ -390,6 +406,194 @@ module.exports = {
...
+      },
+      {
+        name: 'interval',
+        props: VTimestampScopedProps
+      }
+    ],
+    functions: [
+      {
+        name: 'weekdayFormat',
+        signature: '(timestamp: {date: string, time: string, year: number, month: number, day: number, hour: number, minute: number, weekday: number, hasDay: boolean, hasTime: boolean, past: boolean, present: boolean, future: boolean}, short: boolean): string'
```
Comment (nekosaur) [inline packages/api-generator/src/map.js]: Is signature identical for these? If so pull it out into a variable.

Comment (sh7dm) [reply packages/api-generator/src/map.js]: @nekosaur looks like not

**Comment by ClickerMonkey** [comment]
TODO:

- Add scroll push functionality
- Add utility function available in interval and dayBody slots which converts a time or VTimestamp into a pixel value on the y-axis
- Update documentation explaining dayBody slot

**Comment by scottpage** [comment]
Will this receive support for scheduling resources, with timeline view?  i.e. People, rooms, etc.
I'm currently using https://github.com/CroudTech/vue-fullcalendar but it depends on https://fullcalendar.io/

Great work, BTW, thanks for doing this.

Example:
timeline-view

**Pull request merged**